src/generate-grpc-web.ts

```diff
@@ -150,30 +159,42 @@ export function addGrpcWebMisc(options: Options, _file: FileSpec): FileSpec {
   file = file.addCode(
     CodeBlock.empty()
       .addStatement('import UnaryMethodDefinition = grpc.UnaryMethodDefinition')
-      .addStatement('type UnaryMethodDefinitionish = UnaryMethodDefinition<any, any>')
+      .addStatement('interface UnaryMethodDefinitionishR extends UnaryMethodDefinition<any, any> { requestStream: any; responseStream: any; }')
+      .addStatement('type UnaryMethodDefinitionish = UnaryMethodDefinitionishR')
   );
-  file = file.addInterface(generateGrpcWebRpcType());
-  file = file.addClass(generateGrpcWebImpl());
+  file = file.addInterface(generateGrpcWebRpcType(options.returnObservable));
+  file = file.addClass(options.returnObservable ? generateGrpcWebImplObservable() : generateGrpcWebImplPromise());
   return file;
 }
 
-/** Makes an `Rpc` interface to decouple from the low-level grpc-web `grpc.unary`/etc. methods. */
-function generateGrpcWebRpcType(): InterfaceSpec {
+/** Makes an `Rpc` interface to decouple from the low-level grpc-web `grpc.invoke and grpc.unary`/etc. methods. */
+function generateGrpcWebRpcType(returnObservable: boolean): InterfaceSpec {
   let rpc = InterfaceSpec.create('Rpc');
-  let fn = FunctionSpec.create('unary');
+  let fnU = FunctionSpec.create('unary');
+  let fnI = FunctionSpec.create('invoke');
   const t = TypeNames.typeVariable('T', TypeNames.bound('UnaryMethodDefinitionish'));
-  fn = fn
+  fnU = fnU
+    .addTypeVariable(t)
+    .addParameter('methodDesc', t)
+    .addParameter('request', TypeNames.ANY)
+    .addParameter('metadata', TypeNames.unionType(TypeNames.anyType('grpc.Metadata'), TypeNames.UNDEFINED))
+    .returns(
+      returnObservable
+        ? TypeNames.anyType('Observable@rxjs').param(TypeNames.ANY)
+        : TypeNames.PROMISE.param(TypeNames.ANY)
+    );
+  fnI = fnI
     .addTypeVariable(t)
     .addParameter('methodDesc', t)
     .addParameter('request', TypeNames.ANY)
     .addParameter('metadata', TypeNames.unionType(TypeNames.anyType('grpc.Metadata'), TypeNames.UNDEFINED))
-    .returns(TypeNames.PROMISE.param(TypeNames.ANY));
-  rpc = rpc.addFunction(fn);
+    .returns(TypeNames.anyType('Observable@rxjs').param(TypeNames.ANY));
```

Comment:
I think technically this is going to add an `rxjs` dependency for anyone using the grpc-web output, i.e. even if they don't use streaming at all, their `Rpc` interface is going to have this `invoke` type in it...

Can we conditionally add the `invoke` method, and only output if we know they have methods that are streaming?